Implement Mayhem

nottirb opened the pull request:
- Added a `Dockerfile` for fuzz testing
- Added `Mayhemfile(s)` for fuzz testing
- Added a `mayhem.yml` file for automatically running fuzz tests with Mayhem using the mcode action
- Wrote 2 fuzz harnesses

nottirb committed:
Implement Mayhem

nottirb committed:
Cleanup

nottirb committed:
Add Image Tag

nottirb committed:
Fix Errors

nottirb committed:
Cleanup

nottirb committed:
Reduce Test Suite Size

xansec commented on fuzz/fuzz_targets/lerp.rs:
```diff
@@ -0,0 +1,32 @@
+use honggfuzz::fuzz;
```
Can you explain a bit what's going on here? In general we want to avoid non-determinism as it makes reproducing bugs much harder (the same random number won't be used and the bug may not reproduce. I do see that you seed the random number from the test case itself, so this may be ok, but would like to hear if this is guaranteed in StdRng (see https://stackoverflow.com/questions/62309424/does-numpy-random-seed-always-give-the-same-random-number-every-time for commentss).

nottirb replied:
StdRng is guaranteed to generate the same random number when ran on the same build/platform. I'll swap this to Pcg64 to ensure that it generates the same random number on all builds/platforms.

xansec commented on fuzz/fuzz_targets/splines.rs:
```diff
@@ -0,0 +1,133 @@
+use honggfuzz::fuzz;
```
Same question here - can we guarantee the same rand will be generated each time?

xansec commented on mayhem/lerp.Mayhemfile:
```diff
@@ -0,0 +1,10 @@
+project: splines
```
Please omit image name.

xansec commented on mayhem/splines.Mayhemfile:
```diff
@@ -0,0 +1,10 @@
+project: splines
```
Please omit image name.

xansec reviewed (changes requested):
Please make the requested changes.

nottirb committed:
Swap out StdRng for Pcg64

nottirb committed:
Remove Images

nottirb commented:
I removed the image name from the Mayhemfiles, and replaced StdRng with Pcg64 to ensure that it generates the same random number on different builds/platforms.

xansec reviewed (approved):
LGTM!
